users/views.py

```diff
@@ -91,4 +91,34 @@ def post(self, request):
             return JsonResponse({'message':'INVALID_PASSWORD'}, status=401)
 
         except KeyError:
-            return JsonResponse({'message':'KEY_ERROR'}, status=400)
\ No newline at end of file
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)
+
+
+class SocialLoginView(View):
+    def post(self, request):
+        try:
+            access_token = request.headers['Authorization']
+            url          = 'https://kapi.kakao.com/v2/user/me'
+            headers      = {
+                'Authorization' : f'Bearer {access_token}',
+                'Content-type'  : 'application/x-www-form-urlencoded;charset=utf-8'
+            }
+
+            response        = requests.get(url, headers=headers)
+            social_response = response.json()
+
+            if social_response.get('code') == -401:
+                return JsonResponse({'message':'INVALID_ACCESS_TOKEN'}, status=401)
+            
+            if not SocialUser.objects.filter(social_user=social_response['id']).exists():
+                SocialUser(
+                    social_user = social_response['id']
+                ).save()
+
+            token = jwt.encode({'id':SocialUser.objects.get(social_user=social_response['id']).id}, SECRET_KEY, algorithm='HS256')
+            return JsonResponse({'token':token},status=200)
```

Comment:
return 해주는 token이 명확히 어떤 token인지 알려주는게 좋습니다.
나중에는 `refresh token`이나 `access token` 등 여러가지 토큰을 관리해야되는 경우를 고려해야됩니다~
```suggestion
            return JsonResponse({'access_token':token},status=200)
```

Reply:
front로부터 access_token을 받아 카카오에서 유저의 정보를 가져오고, 해당 id로 저희 사이트에서 만든 token을 생성해 return하고 싶었는데요 이때도 access_token에 해당하는 걸까용?

Reply:
네 연우님! 정확하게는 프론트에서 `kakao의 access token`을 서버가 받아서 우리 server의 secret key로 생성한 access_token을 발급해주는 것입니다. 그렇기 때문에 우리가 발급해준 토큰 역시 우리 서버의 자원에 접근하기 위한 access_token 입니다.

Reply:
아하! 감사합니다 병민님!!! 항상 그냥 token으로만 썼었는데 앞으로 access_token이라고 정확히 써줘야겠네요~ PR 다시 올리겠습니다~!